### DISCUSSION TITLE
M5Stack PoEsp32

### DISCUSSION THREAD

**Discussion by Beetix** [arendst/Tasmota#17794] [Q&A] [answered]
Hi,

I am trying to get Tasmota running on a M5Stack PoEsp32 (a device on which I already managed to flash ESPHome). It only has Ethernet connectivity so I've been trying to compile a firmware with the template included. I used https://templates.blakadder.com/olimex_ESP32-POE.html and https://esphome.io/components/ethernet.html?highlight=ethernet#configuration-examples as reference. The firmware runs but the Ethernet does not seem configured (no led blinking on the port).

Here is the content of my user_config_override.h:
```c
#ifdef MODULE
#undef MODULE
#endif
#define MODULE USER_MODULE

#ifdef FALLBACK_MODULE
#undef FALLBACK_MODULE
#endif
#define FALLBACK_MODULE USER_MODULE

#ifdef USER_TEMPLATE
#undef USER_TEMPLATE
#endif
#define USER_TEMPLATE "{\"NAME\":\"M5Stack PoEsp32\",\"GPIO\":[0,0,0,0,0,5536,0,0,0,0,0,0,0,210,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],\"FLAG\":0,\"BASE\":1}"
 
#define USER_RULE1 "ON system#boot DO wifi 0 ENDON"
#define USER_BACKLOG "Rule1 1"

#ifndef USE_ETHERNET
#define USE_ETHERNET
#endif

#ifdef ETH_TYPE
#undef ETH_TYPE
#endif
#define ETH_TYPE          1                    // [EthType] 0 = ETH_PHY_LAN8720, 1 = ETH_PHY_TLK110/ETH_PHY_IP101, 2 = ETH_PHY_RTL8201, 3 = ETH_PHY_DP83848, 4 = ETH_PHY_DM9051, 5 = ETH_PHY_KSZ8081

#ifdef ETH_CLKMODE
#undef ETH_CLKMODE
#endif
#define ETH_CLKMODE       0                    // [EthClockMode] 0 = ETH_CLOCK_GPIO0_IN, 1 = ETH_CLOCK_GPIO0_OUT, 2 = ETH_CLOCK_GPIO16_OUT, 3 = ETH_CLOCK_GPIO17_OUT

#ifdef ETH_ADDRESS
#undef ETH_ADDRESS
#endif
#define ETH_ADDRESS       1                    // [EthAddress] 0 = PHY0 .. 31 = PHY31
```
This is what I get on the serial:
```
00:00:00.002 HDW: ESP32-D0WD
00:00:00.035 UFS: FlashFS mounted with 308 kB free
00:00:00.079 CFG: Loaded from File, Compte 8
00:00:00.089 QPC: Count 1
00:00:00.176 BRY: Berry initialized, RAM used=4041 bytes
00:00:00.199 Projet tasmota - Tasmota Version 12.3.1(tasmota)-2_0_5_3(2023-01-26T12:11:55)
00:00:04.557 WIF: WifiManager actif pour 3 minutes
00:00:05.421 HTP: Serveur web actif sur tasmota-13CE80-3712 avec l'adresse IP 192.168.4.1
00:00:11.302 QPC: Reset
```
Does anyone have an idea on why this is not working?

Thanks

**Comment by sfromis** [comment, marked as answer]
The *defaults* you create in `user_config_override.h` are not used for anything if you "just" upgrade from a already initialized Tasmota setup, only during initial install, or if you do a full "factory reset", like via the command `Reset 1` or the button in the web UI. Success depends on which build environment you select, should normally be just "tasmota32", and not some "tasmota32-xxxxxx" environment. 

Actually, you do not even need to do your own build to get it running with the Ethernet setup, as standard builds already include Ethernet support. Then you can customize the template, and do the config via commands `EthType`, `EthAddress` and  `EthClockMode`.

Instead of your rule to always turn off Wifi, I'd recommend this small Berry script:
https://tasmota.github.io/docs/Berry-Cookbook/#ethernet-network-flipper
This allows you to have Wifi as fallback in case of Ethernet being inoperable, but it will also turn off Wifi if booting with Ethernet available, to make Ethernet the preferred connection if available.

**Reply by Beetix** [reply]
The thing is there is no WiFi on this device: [ESP32-WROOM-32U | 240MHz dual core, 600 DMIPS, 520KB SRAM (without integrated 3D antenna, does not support WiFi wireless function)](https://docs.m5stack.com/en/unit/poesp32#specification)

**Reply by sfromis** [reply]
Well, that's just a "defective device" with nothing to fall back on. My preference would still be to not do a custom build, but just make the relevant configs via serial console after installing a precompiled binary like `tasmota32.bin`.

Of course, you could have other good reasons for a custom build, but it is not a requirement to get such a device operational.

One structural difference from ESPHome is that you should not expect a build per device, as most devices can be handled just fine by using one of the precompiled builds.

**Reply by Beetix** [reply]
Understood. For my use case the module [Teleinfo](https://tasmota.github.io/docs/Teleinfo/) which is not included in the standard firmware is needed.

What is troubling me is that the Ethernet is not working even though my config is looking right (also checked using serial)...

**Reply by barbudor** [reply]
Is the template actually loaded and activated ?
As sfromis pointed, it's one thing to have it in the build and another thing to have it activated (in case of an upgrade)
`backlog Template ; module ; gpio all` could give us some hints

**Reply by Beetix** [reply]
Yes it is:
```
00:03:38.781 CMD: Template
00:03:38.786 RSL: RESULT = {"NAME":"M5Stack PoEsp32","GPIO":[0,0,0,0,0,5536,0,0,0,0,0,0,0,210,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}
```

**Reply by barbudor** [reply]
Thats only one third on the expected answer

The template is loaded, but it doesn't prove it is activated
That's what `Module` would tell us (is answer is 0)
And `gpio all` would also confirm the gpio assignements

**Reply by sfromis** [reply]
Also verify the values of  `Backlog EthType; EthAddress; EthClockMode` (you can copy/paste as one single command)

**Reply by Beetix** [reply]
Everything checks out (I've changed GPIO17 and BASE since last log)
```
00:01:22.570 CMD: backlog Template ; module ; gpio all
00:01:22.619 RSL: RESULT = {"NAME":"M5Stack PoEsp32","GPIO":[0,0,0,0,0,5536,0,0,0,0,0,0,0,0,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":18}
00:01:22.819 RSL: RESULT = {"Module":{"0":"M5Stack PoEsp32"}}
00:01:23.032 RSL: RESULT = {"GPIO0":{"0":"Aucun"},"GPIO1":{"0":"Aucun"},"GPIO2":{"0":"Aucun"},"GPIO3":{"0":"Aucun"},"GPIO4":{"0":"Aucun"},"GPIO5":{"5536":"ETH POWER"},"GPIO6":{"0":"Aucun"},"GPIO7":{"0":"Aucun"},"GPIO8":{"0":"Aucun"},"GPIO9":{"0":"Aucun"},"GPIO10":{"0":"Aucun"},"GPIO11":{"0":"Aucun"},"GPIO12":{"0":"Aucun"},"GPIO13":{"0":"Aucun"},"GPIO14":{"0":"Aucun"},"GPIO15":{"0":"Aucun"},"GPIO16":{"0":"Aucun"},"GPIO17":{"0":"Aucun"},"GPIO18":{"5600":"ETH MDIO"},"GPIO19":{"0":"Aucun"},"GPIO20":{"0":"Aucun"},"GPIO21":{"0":"Aucun"},"GPIO22":{"0":"Aucun"},"GPIO23":{"5568":"ETH MDC"},"GPIO24":{"0":"Aucun"},"GPIO25":{"0":"Aucun"},"GPIO26":{"0":"Aucun"},"GPIO27":{"0":"Aucun"},"GPIO32":{"0":"Aucun"},"GPIO33":{"0":"Aucun"},"GPIO34":{"0":"Aucun"},"GPIO35":{"0":"Aucun"},"GPIO36":{"0":"Aucun"},"GPIO37":{"0":"Aucun"},"GPIO38":{"0":"Aucun"},"GPIO39":{"0":"Aucun"}}
00:02:49.619 CMD: Backlog EthType; EthAddress; EthClockMode
00:02:49.661 RSL: RESULT = {"EthType":1}
00:02:49.864 RSL: RESULT = {"EthAddress":1}
00:02:50.068 RSL: RESULT = {"EthClockMode":0}
```

**Reply by Beetix** [reply]
Flashed the device using the Web Installer with precompiled firmware Tasmota Teleinfo. Then sent the Ethernet config (EthAddress, EthClockMode and EthType), template and module through serial. Plugged the M5Stack PoESP32 on a PoE switch and it worked!

I don't know why the custom firmware didn't work...

The downside of this method is that I need to send some config over serial. This means I can't switch from ESPHome to Tasmota through OTA...

**Comment by keshavdv** [comment]
The following template worked for me and I'm able to successfully connect over ethernet (flashing seemed to require overriding the baud rate to 115200):

```
#ifdef MODULE
#undef MODULE
#endif
#define MODULE USER_MODULE

#ifdef FALLBACK_MODULE
#undef FALLBACK_MODULE
#endif
#define FALLBACK_MODULE USER_MODULE

#ifdef USER_TEMPLATE
#undef USER_TEMPLATE
#endif
#define USER_TEMPLATE "{\"NAME\":\"M5Stack PoEsp32\",\"GPIO\":[1,1,1,1,1,1,0,0,5536,1,1,1,1,0,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,1,1,1,1,1,0,0,1],\"FLAG\":0,\"BASE\":1}"

#define USER_RULE1 "ON system#boot DO wifi 0 ENDON"
#define USER_BACKLOG "Rule1 1"

#ifndef USE_ETHERNET
#define USE_ETHERNET
#endif

#ifdef ETH_TYPE
#undef ETH_TYPE
#endif
#define ETH_TYPE          1                    // [EthType] 0 = ETH_PHY_LAN8720, 1 = ETH_PHY_TLK110/ETH_PHY_IP101, 2 = ETH_PHY_RTL8201, 3 = ETH_PHY_DP83848, 4 = ETH_PHY_DM9051, 5 = ETH_PHY_KSZ8081

#ifdef ETH_CLKMODE
#undef ETH_CLKMODE
#endif
#define ETH_CLKMODE       0                    // [EthClockMode] 0 = ETH_CLOCK_GPIO0_IN, 1 = ETH_CLOCK_GPIO0_OUT, 2 = ETH_CLOCK_GPIO16_OUT, 3 = ETH_CLOCK_GPIO17_OUT

#ifdef ETH_ADDRESS
#undef ETH_ADDRESS
#endif
#define ETH_ADDRESS       1                    // [EthAddress] 0 = PHY0 .. 31 = PHY31
```

**Reply by Beetix** [reply]
Hum very strange... You really took the same GPIO mapping as the [Olimex ESP32-POE](https://templates.blakadder.com/olimex_ESP32-POE.html) and it worked on the M5Stack PoESP32?

**Reply by Beetix** [reply]
The main difference I see is that ETH POWER is on GPIO12 for the Olimex ESP32-POE and on GPIO05 for M5Stack PoESP32

**Reply by keshavdv** [reply]
Ah, oops, I realize I had set the template manually via serial to the following and my paste is wrong:
```
{"NAME":"M5Stack PoESP32","GPIO":[0,0,0,0,0,5536,0,0,0,0,0,0,1,1,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}
```

I did have to power cycle the device after applying this template and flashing

**Reply by Beetix** [reply]
Ok, same pins for ethernet then. I guess I'll give it another try tomorrow :sweat_smile:

**Reply by Beetix** [reply]
Still no success... @keshavdv which version of Tasmota are you using?

**Reply by shikousa** [reply]
@Beetix Could you make the M5Stack ESP32 Ethernet unit working?
What is the correct template now?

**Reply by Beetix** [reply]
@shikousa yes, I used a precompiled firmware and configured the module over serial.
The template is the following:
```
{"NAME":"M5Stack PoEsp32","GPIO":[0,0,0,0,0,5536,0,0,0,0,0,0,0,0,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":18}
```

**Reply by Wupsman** [reply]
@Beetix, thank you for opening this discussion!

So far everything has worked directly from Visual Studio Code with the customized Tasmota firmware, thanks to your help.

Now I just have to manage to read my electricity meter from the grid ;-)

` #ifdef` MODULE
#undef MODULE
#endif
#define MODULE USER_MODULE

#ifdef FALLBACK_MODULE
#undef FALLBACK_MODULE
#endif
#define FALLBACK_MODULE USER_MODULE

#ifdef USER_TEMPLATE
#undef USER_TEMPLATE
#endif
#define USER_TEMPLATE "{\"NAME\":\"M5Stack PoEsp32\",\"GPIO\":[0,0,0,0,0,5536,0,0,0,0,0,0,1,1,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],\"FLAG\":0,\"BASE\":1}"

#ifndef USE_ETHERNET
#define USE_ETHERNET
#endif

#ifdef ETH_TYPE
#undef ETH_TYPE
#endif
#define ETH_TYPE          1                    // [EthType] 0 = ETH_PHY_LAN8720, 1 = ETH_PHY_TLK110/ETH_PHY_IP101, 2 = ETH_PHY_RTL8201, 3 = ETH_PHY_DP83848, 4 = ETH_PHY_DM9051, 5 = ETH_PHY_KSZ8081

#ifdef ETH_CLKMODE
#undef ETH_CLKMODE
#endif
#define ETH_CLKMODE       0                    // [EthClockMode] 0 = ETH_CLOCK_GPIO0_IN, 1 = ETH_CLOCK_GPIO0_OUT, 2 = ETH_CLOCK_GPIO16_OUT, 3 = ETH_CLOCK_GPIO17_OUT

#ifdef ETH_ADDRESS
#undef ETH_ADDRESS
#endif
#define ETH_ADDRESS       1                    // [EthAddress] 0 = PHY0 .. 31 = PHY31

#ifndef USE_SCRIPT
#define USE_SCRIPT                               // Add support for script (+17k code)
#endif
#ifndef USE_SML_M
#define USE_SML_M
#endif
#ifdef USE_RULES
#undef USE_RULES
#endif `

**Comment by shikousa** [comment]
Thanks. The 4-pin plug of the M5Stack PoEsp32 gives you GND, 5V, RX (GPIO16) and TX (GPIO17). The 5V is provided by POE. How can I use the 5V for RX/TX as they are actually expecting 3,3V?

https://docs.m5stack.com/en/unit/poesp32

**Comment by em24tom** [comment]
Hi and sorry to ask this, but I'm new to tasmota and ESP32....

I have flashed tasmota on the M5 stack POESP32 ( VSC and PIO ) and now I am trying to "...configure the module via the serial port..." like you did

> @shikousa yes, I used a precompiled firmware and **configured the module over serial**. The template is the following:
> 

unfortunately I'm too stupid for that...

Is there a step-by-step guide somewhere I can dive into?

The first thing I need to do is: switch on the wired Ethernet and specify a fixed IP configuration.

hopefully this will get me to the tasmota interface and the rest will be self-explaning

thanks for helping me and have a nice day
greetings tom

**Comment by Wupsman** [comment]
Hi [em24tom](https://github.com/em24tom)

starting with this device with Tasmota is probably a bit difficult. Have you tried a device listed at []https://templates.blakadder.com/ ?

When I wanted to take the M5 into operation for the first time, I was a bit surprised that I had to disassemble it - but the info from the hassio community helped me.
[]https://community.home-assistant.io/t/m5stack-poecam-esp32-board-support/495489/5

**Comment by em24tom** [comment]
Hi Wupsman,

you are making me a little nervous, but what would be better about using a listed device?
I will flash it, everything works fine, nothing learned.
...and I don't have any such device here...
If i really can't do it without wifi, i'll try the WT32-ETH01. It's already waiting on the desk and i have found instructions and also YT videos for this.
But the goal should be the PoESP32 from M5. Because wifi in the meter cupboard is not an option for me and many others.
Simply connect POE to the M5 and at serial the orignal Hichi Smartmeter reader to grab SML, no wifi and no extra power supply.

So i still need to get the wired ethernet on und connect this way to the tasmota web-interface.

When everything works: write a manual for everyone and report a template here.

tom

**Reply by sfromis** [reply]
The benefit of using a device already "known to Tasmota" is that someone else has done the work to create a working setup. While getting Tasmota up and running should be pretty simple, you do need the right config settings to match the Ethernet hardware.
https://tasmota.github.io/docs/Commands/#ethernet
Until Ethernet is configured, you can use Wifi or serial access.

**Reply by Wupsman** [reply]
Hi [em24tom](https://github.com/em24tom)

it works, put the folow in your user_config_override:

`#ifdef MODULE
#undef MODULE
#endif
#define MODULE USER_MODULE

#ifdef FALLBACK_MODULE
#undef FALLBACK_MODULE
#endif
#define FALLBACK_MODULE USER_MODULE

#ifdef USER_TEMPLATE
#undef USER_TEMPLATE
#endif
#define USER_TEMPLATE "{"NAME":"M5Stack PoEsp32","GPIO":[0,0,0,0,0,5536,0,0,0,0,0,0,1,1,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}"

#ifndef USE_ETHERNET
#define USE_ETHERNET
#endif

#ifdef ETH_TYPE
#undef ETH_TYPE
#endif
#define ETH_TYPE 1 // [EthType] 0 = ETH_PHY_LAN8720, 1 = ETH_PHY_TLK110/ETH_PHY_IP101, 2 = ETH_PHY_RTL8201, 3 = ETH_PHY_DP83848, 4 = ETH_PHY_DM9051, 5 = ETH_PHY_KSZ8081

#ifdef ETH_CLKMODE
#undef ETH_CLKMODE
#endif
#define ETH_CLKMODE 0 // [EthClockMode] 0 = ETH_CLOCK_GPIO0_IN, 1 = ETH_CLOCK_GPIO0_OUT, 2 = ETH_CLOCK_GPIO16_OUT, 3 = ETH_CLOCK_GPIO17_OUT

#ifdef ETH_ADDRESS
#undef ETH_ADDRESS
#endif
#define ETH_ADDRESS 1 // [EthAddress] 0 = PHY0 .. 31 = PHY31

#ifndef USE_SCRIPT
#define USE_SCRIPT // Add support for script (+17k code)
#endif
#ifndef USE_SML_M
#define USE_SML_M
#endif
#ifdef USE_RULES
#undef USE_RULES
#endif `

If you want to read an electricity meter, I would use AMS Reader if your meter is supported.

[gskjold](https://github.com/gskjold) has published a prerelease firmware for the PoEsp32 here:
[amsreader](https://github.com/UtilitechAS/amsreader-firmware/issues/365#issuecomment-1695288043)

**Comment by em24tom** [comment]
Thank you very mouch!
That's exactly what I found out and added to my to-do list step by step
but it still doesn't work?!

why does the template in the "user_config_override.h" hase to be changed to this?

`#define USER_TEMPLATE "{\"NAME\":\"M5Stack PoEsp32\",\"GPIO\":[0,0,0,0,0,5536,0,0,0,0,0,0,1,1,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],\"FLAG\":0,\"BASE\":1}"
`

What I mean are all these `\` Backslashes...
What are they doing?

**Reply by sfromis** [reply]
Escaping `"` using a backslash is needed in the C++ syntax, as the whole string is wrapped in `"`, meaning that using `"` within the string would not work. Hence the syntax is that you can still include `"` in strings using the escape character, `\"` - this is also used to include things like tab `\t` or newline `\n`.

**Comment by em24tom** [comment]
that is convincing
Thank you very much!

**Comment by em24tom** [comment]
so far, so good... it runs the first time

unfortunately i can't force the esp32 to have a fixed ip after flashing

i found this cmd and it works on console, but where do i place this for a fixed ip directly after flashing?
`EthIPAddress 192.168.178.27`
`EthGateway 192.168.178.1`
`EthSubnetmask 255.255.255.0`
`EthDNSServer1 192.168.178.65`

**Reply by em24tom** [reply]
do i have to turn wifi off first? ( to turn off DHCP )

but the one in user_config_override.h has no effect:

`#define USER_RULE1 "ON system#boot DO wifi 0 ENDON"`
`#define USER_BACKLOG "Rule1 1"`

**Reply by sfromis** [reply]
Whatever you do to include commands in a binary will not take effect if you just upgrade to the binary, only if there is done a full reset. And the trigger `System#boot` will only fire **after** network connection.

**Reply by em24tom** [reply]
Sorry for the misunderstanding
of course I would like to have a static IP even after the reset

So I connect to my Ethernet, go to the tasmota web interface and call up the console
Here I enter: "Reset 1"
The PoEsp32 restarts... and logs in again with DHCP on the wired Ethernet? ( ip 192.168.178.184 )

00:00:00.002 HDW: ESP32-D0WD v1.0 
00:00:00.049 UFS: FlashFS mounted with 300 kB free
00:00:00.075 CFG: Loaded from File, Count 2
00:00:00.085 QPC: Count 1
00:00:00.091 SCR: ufilesystem found
00:00:00.623 BRY: Berry initialized, RAM used 5180 bytes
00:00:00.645 Project tasmota - Tasmota Version 13.4.0(tasmota32)-2_0_14(2024-03-25T19:00:12)
00:00:03.099 ETH: Connected at 100Mbps Full Duplex, Mac B8:D6:1A:70:7E:E7, Hostname tasmota-707EE4-7908-eth
22:15:47.112 WIF: WifiManager active for 3 minutes
22:15:48.316 HTP: Web server active on tasmota-707EE4-7908-eth with IP address 192.168.178.184
22:15:48.330 HTP: Web server active on tasmota-707EE4-7908-eth with IP address 192.168.178.184
22:15:49.381 RSL: INFO1 = {"Info1":{"Module":"M5Stack PoEsp32","Version":"13.4.0(tasmota32)","FallbackTopic":"cmnd/DVES_707EE4_fb/","GroupTopic":"cmnd/tasmotas/"}}
22:15:49.395 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Ethernet":{"Hostname":"tasmota-707EE4-7908-eth","IPAddress":"192.168.178.184"}}}
22:15:49.408 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":1}}
22:15:53.437 RSL: STATE = {"Time":"2024-03-25T22:15:53","Uptime":"0T00:00:07","UptimeSec":7,"Heap":129,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":4,"Objects":46}}
22:15:54.441 QPC: Reset

Where is the right place to force the static IP address 192.168.178.27?

**Reply by sfromis** [reply]
There is no **the** right place, as usual there are many ways of doing things.

Since you found USER_BACKLOG, that is a workable place to issue commands, which are executed for first boot after a full reset. For that to work, you should probably also make sure to include the template in the build.

**Comment by Jonn3y** [comment]
Can you guys help me here, i also try to install Tasmota to PoESP32 but after flashing the Device is in a boot loop and i got multple different errors after flashing:
First:
```
[11:42:53]rst:0x3 (SW_RESET),boot:0x12 (SPI_FAST_FLASH_BOOT)
[11:42:53]configsip: 0, SPIWP:0xee
[11:42:53]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[11:42:53]mode:DIO, clock div:2
[11:42:53]load:0x3fff0030,len:1184
[11:42:53]load:0x40078000,len:12784
[11:42:53]load:0x40080400,len:3032
[11:42:53]entry 0x400805e4
[11:42:53]ets Jun  8 2016 00:22:57
[11:42:53]
```

Second:
```
11:53:16.761 -> rst:0x10 (RTCWDT_RTC_RESET),boot:0x12 (SPI_FAST_FLASH_BOOT)
11:53:16.761 -> flash read err, 1000
11:53:16.761 -> ets_main.c 371 
11:53:17.134 -> ets Jun  8 2016 00:22:57
11:53:17.134 -> 
11:53:17.134 -> rst:0x10 (RTCWDT_RTC_RESET),boot:0x12 (SPI_FAST_FLASH_BOOT)
11:53:17.174 -> flash read err, 1000
```

And now finally:
```
rst:0xc (SW_CPU_RESET),boot:0x12 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:13228
ho 0 tail 12 room 4
load:0x40080400,len:4
load:0x40080404,len:2924
entry 0x40080570
ets Jun  8 2016 00:22:57
```

Tried using https://tasmota.github.io/install/ and ESP Flasher, a precompiled Tasmota and also compiled one myself as mentioned above. 
Serial Adapter is a CH340
CH340<->ESP32
RX<->TX
TX<->RX
3.3V<->3.3V
GND<->GND
GND<->G0
I did not use the EN Pin and also tried to flash with and without the UART/PoE? Board connected. No success.

The device itself seemed to be fine. I pluged it into the Network and have seen it pulling an IP with original firmware in the beginning. But as stupid as i am i did not make an backup first.

Greetings.

**Reply by sfromis** [reply]
Well, you did not tell what *specific* binary you flashed. While the Tasmota Web Installer (which is the recommended tool) will automatically pick a `.factory` binary, other ways means that you have to make sure of this.

One possible source of trouble is also if you are not supplying good power to the device, especially when ESP32, and many USB/Serial adapters are not up to this. Saying "a CH340" is no details of which one.

**Reply by Jonn3y** [reply]
Tried the "Tasmota32 Bluetooth (english)" and also the "Tasmota (english)"  version using the Web installer.
I dont know how to tell which Serial Adapter it is exactly, but i never had issues with this one with multiple ESP8266.
It looks like this 
![s-l16002](https://github.com/user-attachments/assets/56618a18-3137-4c3f-9181-c3f99414c91d)

I also tried using the esptool and set the baud rate to 115200, but that also did not help. Feels a little like either the Serial controller or the device misbehave here.

**Reply by Jonn3y** [reply]
Well, somehow i might have been stupid. I recompiled Tasmota 14 with the additional template etc mentioned above. Uploaded the tasmota32.factory.bin it to the Web Intaller, rebooted the device. Then i got an error in the console:
```
[15:57:32]rst:0xc (SW_CPU_RESET),boot:0x12 (SPI_FAST_FLASH_BOOT)
[15:57:32]configsip: 0, SPIWP:0xee
[15:57:32]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[15:57:32]mode:DIO, clock div:2
[15:57:32]load:0x3fff0030,len:184
[15:57:32]load:0x40078000,len:13228
[15:57:32]ho 0 tail 12 room 4
[15:57:32]load:0x40080400,len:4
[15:57:32]load:0x40080404,len:2924
[15:57:32]entry 0x40080570
```

Just umplugged the Serial Adapter and plugged in the Ethernet Cable and eventually the LEDs started blinking. Finally. Tasmota is working now and is reachable over PoE

**Reply by sfromis** [reply]
Yeah, that USB/serial adapter is way too wimpy to power ESP32, where a better one is needed, like one [recommended](https://tasmota.github.io/docs/Getting-Started/#serial-programmer) by the project.
